src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Primitives/ComposablePartDefinition.cs

```diff
@@ -140,10 +140,12 @@ internal virtual bool TryGetExports(ImportDefinition definition, [NotNullWhen(tr
             if (multipleExports != null)
             {
                 multipleMatches = multipleExports;
+                // TODO2 singleMatch dould be null when returning true
+                singleMatch = null!;
```

Comment:
:memo: the `[NotNullWhen(...)]` annotation on method seems incorrect

Reply:
Yes, please remove it.  Thanks.